src/asDate.ts

```diff
@@ -1,5 +1,16 @@
 import type { DateField, TimestampField } from "@prismicio/types";
 
+/**
+ * The return type of `asDate()`.
+ */
+type AsDateReturnType<
+	Field extends DateField | TimestampField | null | undefined,
```

Comment:
What do you think of moving `null | undefined` out from the generic and into the parameter type (i.e. `dateOrTimestampField: Field | null | undefined`)? It keeps the `Field` generic strictly to a Prismic field.

There might be type intricacies that I missed so maybe this placement is on purpose.

Reply:
It it's not a generic we cannot send it to the extend in the return type because you cannot run `typeof field` there :/

Reply:
Ah gotcha! Okay, no worries.